source/pkg/reconciler/resources/receive_adapter.go

```diff
@@ -0,0 +1,181 @@
+	replicas := int32(1)
+
+	env := []corev1.EnvVar{
+		{
+			Name:  "RABBITMQ_BROKERS",
+			Value: args.Source.Spec.Brokers,
+		},
+		{
+			Name:  "RABBITMQ_TOPIC",
+			Value: args.Source.Spec.Topic,
+		},
+		{
+			Name: "RABBITMQ_USER",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: args.Source.Spec.User.SecretKeyRef,
+			},
+		},
+		{
+			Name: "RABBITMQ_PASSWORD",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: args.Source.Spec.Password.SecretKeyRef,
+			},
+		},
+		{
+			Name:  "RABBITMQ_ROUTING_KEY",
+			Value: args.Source.Spec.QueueConfig.RoutingKey,
+		},
+		{
+			Name:  "RABBITMQ_CHANNEL_CONFIG_PREFETCH_COUNT",
+			Value: strconv.Itoa(args.Source.Spec.ChannelConfig.PrefetchCount),
+		},
+		{
+			Name:  "RABBITMQ_CHANNEL_CONFIG_QOS_GLOBAL",
+			Value: strconv.FormatBool(args.Source.Spec.ChannelConfig.GlobalQos),
+		},
+		{
+			Name:  "RABBITMQ_EXCHANGE_CONFIG_NAME",
+			Value: args.Source.Spec.ExchangeConfig.Name,
+		},
+		{
+			Name:  "RABBITMQ_EXCHANGE_CONFIG_TYPE",
+			Value: args.Source.Spec.ExchangeConfig.TypeOf,
+		},
+		{
+			Name: "RABBITMQ_EXCHANGE_CONFIG_DURABLE",
+			Value: strconv.FormatBool(args.Source.Spec.ExchangeConfig.Durable),
+		},
+		{
+			Name: "RABBITMQ_EXCHANGE_CONFIG_AUTO_DELETED",
+			Value: strconv.FormatBool(args.Source.Spec.ExchangeConfig.AutoDeleted),
+		},
+		{
+			Name: "RABBITMQ_EXCHANGE_CONFIG_INTERNAL",
+			Value: strconv.FormatBool(args.Source.Spec.ExchangeConfig.Internal),
+		},
+		{
+			Name: "RABBITMQ_EXCHANGE_CONFIG_NOWAIT",
+			Value: strconv.FormatBool(args.Source.Spec.ExchangeConfig.NoWait),
+		},
+		{
+			Name: "RABBITMQ_QUEUE_CONFIG_NAME",
+			Value: args.Source.Spec.QueueConfig.Name,
+		},
+		{
+			Name: "RABBITMQ_QUEUE_CONFIG_DURABLE",
+			Value: strconv.FormatBool(args.Source.Spec.QueueConfig.Durable),
+		},
+		{
+			Name: "RABBITMQ_QUEUE_CONFIG_AUTO_DELETED",
+			Value: strconv.FormatBool(args.Source.Spec.QueueConfig.DeleteWhenUnused),
+		},
+		{
+			Name: "RABBITMQ_QUEUE_CONFIG_EXCLUSIVE",
+			Value: strconv.FormatBool(args.Source.Spec.QueueConfig.Exclusive),
+		},
+		{
+			Name: "RABBITMQ_QUEUE_CONFIG_NOWAIT",
+			Value: strconv.FormatBool(args.Source.Spec.QueueConfig.NoWait),
+		},
+		{
+			Name: "SINK_URI",
+			Value: args.SinkURI,
+		},
+		{
+			Name:  "K_SINK",
+			Value: args.SinkURI,
+		},
+		{
+			Name:  "NAME",
+			Value: args.Source.Name,
+		},
+		{
+			Name:  "NAMESPACE",
+			Value: args.Source.Namespace,
+		},
+		{
+			Name:  "K_LOGGING_CONFIG",
+			Value: args.LoggingConfig,
+		},
+		{
+			Name:  "K_METRICS_CONFIG",
+			Value: args.MetricsConfig,
+		},
+	}
+
+	return &v1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:         utils.GenerateFixedName(args.Source, fmt.Sprintf("rabbitmqsource-%s", args.Source.Name)),
+			Namespace:    args.Source.Namespace,
+			GenerateName: fmt.Sprintf("%s-", args.Source.Name),
+			Labels:       args.Labels,
+			OwnerReferences: []metav1.OwnerReference{
+				*kmeta.NewControllerRef(args.Source),
```

Comment:
if this is the only thing you create then you don't need finalize, ack 👍

Reply:
well, there will also typically be resources created on the RabbitMQ cluster as well (Exchange and Queue, unless they were pre-existing) and the Binding from the Exchange to the Queue. Those need to be cleaned up on deletion as well, and the Binding might need to be replaced if there is an update to source properties that affect it

Reply:
Yeah, that's why I tagged it first above since I was expecting there to be some sort of cleanup necessary, but I must have missed (as I said, just the first pass) on what other cleanup needs to happen. Maybe you could add a comment in the PR at the location where the creation of resources happens that need to get cleaned up, so it won't get lost?